src/univariate/discrete/poisson.jl

```diff
@@ -85,6 +85,19 @@ function entropy(d::Poisson{T}) where T<:Real
 end
 
 
+"""
+    kldivergence(p::Poisson, q::Poisson)
+
+See [KL Poisson](https://en.wikipedia.org/wiki/Poisson_distribution#Other_properties)
+"""
+function kldivergence(p::Poisson, q::Poisson)
+    λp, λq = rate.((p, q))
+    return λq - λp + λp * (log(λp) - log(λq))
+end
+
+
+
+
```

Comment:
Again we should remove broadcasting and reduce some computations. Additionally, `λp` can be zero and hence it is better to use `xlogy` here:
```suggestion
    λp = rate(p)
    λq = rate(q)
    λp_over_λq = λp / λq
    return xlogy(λp, λp_over_λq) - λp_over_λq
end

```

Reply:
Your new formula was incorrect but I still used the `xlogy`

Reply:
Ah true :smile: